BuildConfigGen/Program.cs

```diff
@@ -313,13 +313,36 @@ private static void MainUpdateTask(
 
         private static int GetCurrentSprint()
         {
+            // Scheduled time for Cortesy Push
+            var cortesyPushScheduleDay = DayOfWeek.Tuesday;
+            var cortesyPushUtcTime = new TimeOnly(8, 30); //UTC time
+
             string url = "https://whatsprintis.it";
             var httpClient = new HttpClient();
             httpClient.DefaultRequestHeaders.Add("Accept", "application/json");
 
             string json = httpClient.GetStringAsync(url).Result;
             JsonDocument currentSprintData = JsonDocument.Parse(json);
             int currentSprint = currentSprintData.RootElement.GetProperty("sprint").GetInt32();
+            int week = currentSprintData.RootElement.GetProperty("week").GetInt32();
+
+            if (week == 3) // if it is the end of the current sprint
+            {
+                var nowUtc = DateTime.UtcNow;
+
+                // Increase sprint number if scheduled pipeline was already triggered
```

Comment:
What if scheduled pipeline was failed/delayed?

Reply:
Yes, you're right that it would work incorrectly in this case, but I don't think that it makes sense to add complicated logic there to check the build status

@KonstantinTyukalov, do you have any suggestions for improvement?
Probably, we can leave the comparison for a day of the week to avoid these cases partially because a new build should be ready on Wednesday.